mitiq/calibration/settings.py

```diff
@@ -237,6 +240,14 @@ def print_line(self, performance: str, circuit_type: str) -> None:
 
     def __repr__(self) -> str:
         return str(self.to_dict())
+    
+    def num_circuit_required(self) -> int:
+        summary = self.to_dict()
+        if self.technique is MitigationTechnique.ZNE:
+            return len(summary["scale_factors"])
+        elif self.technique is MitigationTechnique.PEC:
+            return summary["num_samples"]
```

Comment:
To make `mypy` happy, you could add something simple like `return 0` at the end of the function. `mypy` does not know that the only options for `self.technique` are `MitigationTechnique.ZNE` or `MitigationTechnique.PEC`.

```suggestion
    def num_circuit_required(self) -> int:
        summary = self.to_dict()
        if self.technique is MitigationTechnique.ZNE:
            return len(summary["scale_factors"])
        elif self.technique is MitigationTechnique.PEC:
            return summary["num_samples"]
        return 0
```
_**untested!**_

---

BTW, this PR is looking great. I plan to take a deeper look soon. Is it all ready for review?

Reply:
Yes it's ready!